java-tools/OsmAndServer/src/main/java/net/osmand/server/api/services/EmailRegistryService.java

```diff
@@ -6,18 +6,30 @@
 import java.util.Date;
 import java.util.List;
 
-import org.springframework.beans.factory.annotation.Autowired;
+import net.osmand.server.api.repo.OsmRecipientsRepository;
 import org.springframework.jdbc.core.JdbcTemplate;
 import org.springframework.jdbc.core.RowCallbackHandler;
 import org.springframework.jdbc.core.RowMapper;
 import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
 
 @Service
 public class EmailRegistryService {
-
-	@Autowired
-	private JdbcTemplate jdbcTemplate;
-
+	
+	private final JdbcTemplate jdbcTemplate;
+	
+	private final OsmRecipientsRepository recipientsRepository;
+	
+	public EmailRegistryService(OsmRecipientsRepository recipientsRepository, JdbcTemplate jdbcTemplate) {
+		this.recipientsRepository = recipientsRepository;
+		this.jdbcTemplate = jdbcTemplate;
+	}
+	
+	@Transactional
```

Comment:
here transactional not needed - check unsubscribedRepo.deleteAllByEmail(email);

Reply:
I added it after exception No EntityManager. I read it need for deleteBy custom method